### PR TITLE
Using interface_link_up utility in ethtool_test.py [Waiting for utils]

### PR DESCRIPTION
**Pull request by vaishnavibhat**
Using the utility to check if interface's link is up
before proceeding with the ethtool test

Signed-off-by: Vaishnavi Bhat <vaishnavi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by vaishnavibhat** [comment]
Seeing all the test pass if ping count is increased above 25. 

# avocado run ethtool_test.py -m ethtool_test.py.data/ethtool_test.yaml 
JOB ID     : 9e4ce2c28fe224d630b5ae7e1873d6fc005741cf
JOB LOG    : /root/tests/results/job-2019-11-27T08.20-9e4ce2c/job.log
 (01/16) ethtool_test.py:Ethtool.test_ethtool;arg-bf76: PASS (6.13 s)
 (02/16) ethtool_test.py:Ethtool.test_ethtool;Driver-4f63: PASS (5.61 s)
 (03/16) ethtool_test.py:Ethtool.test_ethtool;Autonegotiation-4c73: FAIL: ping failed after interface is up (15.68 s)
 (04/16) ethtool_test.py:Ethtool.test_ethtool;PauseParameter-aecc: PASS (5.57 s)
 (05/16) ethtool_test.py:Ethtool.test_ethtool;Coalescing-cf3f: PASS (5.61 s)
 (06/16) ethtool_test.py:Ethtool.test_ethtool;RegisterDump-fda5: PASS (5.65 s)
 (07/16) ethtool_test.py:Ethtool.test_ethtool;EEPROMDump-249c: PASS (90.60 s)
 (08/16) ethtool_test.py:Ethtool.test_ethtool;RingParameter-8e7b: PASS (5.67 s)
 (09/16) ethtool_test.py:Ethtool.test_ethtool;ShowOffloadProtocolFeature-16af: PASS (5.61 s)
 (10/16) ethtool_test.py:Ethtool.test_ethtool;AdapterInitiate-2d16: PASS (8.66 s)
 (11/16) ethtool_test.py:Ethtool.test_ethtool;Statistics-9579: PASS (5.57 s)
 (12/16) ethtool_test.py:Ethtool.test_ethtool;SelfTest-ccf3: FAIL: ping failed after interface is up (19.24 s)
 (13/16) ethtool_test.py:Ethtool.test_ethtool;Show-permaddr-a928: PASS (5.59 s)
 (14/16) ethtool_test.py:Ethtool.test_ethtool;Get-dump-656e: FAIL: failed (1.49 s)
 (15/16) ethtool_test.py:Ethtool.test_ethtool;Show-time-stamping-90be: PASS (5.64 s)
 (16/16) ethtool_test.py:Ethtool.test_ethtool;Get-phy-tunable-3597: FAIL: failed (1.48 s)
RESULTS    : PASS 12 | ERROR 0 | FAIL 4 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 194.37 s
JOB HTML   : /root/tests/results/job-2019-11-27T08.20-9e4ce2c/results.html


Before:
# avocado run ethtool_test_old.py -m ethtool_test.py.data/ethtool_test.yaml 
JOB ID     : 0d33591442204a85bd4091d1e19397f01d25fca0
JOB LOG    : /root/tests/results/job-2019-11-27T08.28-0d33591/job.log
 (01/16) ethtool_test_old.py:Ethtool.test_ethtool;arg-bf76: FAIL: interface up failed (2.68 s)
 (02/16) ethtool_test_old.py:Ethtool.test_ethtool;Driver-4f63: FAIL: interface up failed (3.00 s)
 (03/16) ethtool_test_old.py:Ethtool.test_ethtool;Autonegotiation-4c73: FAIL: failed (2.99 s)
 (04/16) ethtool_test_old.py:Ethtool.test_ethtool;PauseParameter-aecc: FAIL: interface up failed (2.90 s)
 (05/16) ethtool_test_old.py:Ethtool.test_ethtool;Coalescing-cf3f: FAIL: interface up failed (3.06 s)
 (06/16) ethtool_test_old.py:Ethtool.test_ethtool;RegisterDump-fda5: FAIL: interface up failed (2.98 s)
 (07/16) ethtool_test_old.py:Ethtool.test_ethtool;EEPROMDump-249c: FAIL: interface up failed (88.79 s)
 (08/16) ethtool_test_old.py:Ethtool.test_ethtool;RingParameter-8e7b: FAIL: interface up failed (2.92 s)
 (09/16) ethtool_test_old.py:Ethtool.test_ethtool;ShowOffloadProtocolFeature-16af: FAIL: interface up failed (3.05 s)
 (10/16) ethtool_test_old.py:Ethtool.test_ethtool;AdapterInitiate-2d16: FAIL: interface up failed (6.02 s)
 (11/16) ethtool_test_old.py:Ethtool.test_ethtool;Statistics-9579: FAIL: interface up failed (3.02 s)
 (12/16) ethtool_test_old.py:Ethtool.test_ethtool;SelfTest-ccf3: FAIL: failed (8.00 s)
 (13/16) ethtool_test_old.py:Ethtool.test_ethtool;Show-permaddr-a928: FAIL: interface up failed (2.92 s)
 (14/16) ethtool_test_old.py:Ethtool.test_ethtool;Get-dump-656e: FAIL: failed (2.99 s)
 (15/16) ethtool_test_old.py:Ethtool.test_ethtool;Show-time-stamping-90be: FAIL: interface up failed (2.94 s)
 (16/16) ethtool_test_old.py:Ethtool.test_ethtool;Get-phy-tunable-3597: FAIL: failed (2.98 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 16 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 141.82 s
JOB HTML   : /root/tests/results/job-2019-11-27T08.28-0d33591/results.html

**Comment by vaishnavibhat** [comment]
Depends on utility function:

https://github.com/avocado-framework/avocado/pull/3383

**Comment by narasimhan-v** [comment]
@vaishnavibhat please attach job.log

File: io/net/ethtool_test.py [hunk 1]
```diff
@@ -74,36 +75,32 @@ def interface_state_change(self, interface, state, status):
...
-                                                             .splitlines():
-            if 'Link detected' in line:
-                return line.split()[-1]
-        return ''
-
     def test_ethtool(self):
         '''
         Test the ethtool args provided
         '''
-        for state, status in zip(["down", "up"], ["no", "yes"]):
```
Comment (narasimhan-v) [inline io/net/ethtool_test.py]: @vaishnavibhat we were actually running the test when interface is down and up, does the new approach take that into consideration ?

**Comment by vaishnavibhat** [comment]
Opoened https://github.com/avocado-framework-tests/avocado-misc-tests/pull/1525 to track this issue. There was branch conflict which I couldn't resolve, hence opened a new PR.

File: io/net/ethtool_test.py (hunk 1 above)
Comment (vaishnavibhat) [reply io/net/ethtool_test.py]: Addressed in https://github.com/avocado-framework-tests/avocado-misc-tests/pull/1525

**Pull request closed without merging**